### ISSUE TITLE
Buffing differential generator

### ISSUE THREAD

**Issue by GreatSage777** [Anuken/Mindustry-Suggestions#5072]
### Describe what you would like changed, and why.

I would like to buff differential generator. To make it work from now you have to input 1 pyrotite 3 spores pod and 2.4 cyrofluid liquid input per  second after that it will produce 1080 energy for 7 seconds. Those changes will improve differential generator to use it less recources and produce energy for longer time.

### Describe the changes you want to propose. Include possible alternatives.

Please make differential generator more powerfull.

**Comment by jehosula** [comment]
a more reasonable buff to differential generator this time

**Comment by GreatSage777** [comment]
Thanks for the tip!

пт, 14 черв. 2024, 02:10 користувач jehosula ***@***.***>
пише:

**Comment by jehosula** [comment]
> Thanks for the tip! пт, 14 черв. 2024, 02:10 користувач jehosula ***@***.***> пише:

well this does not say much
adding spore pod input will just make this generator's actual issues even worse
maybe just -2 cryofluid and takes one pyratite every 4 seconds with no spore input?

**Comment by GreatSage777** [comment]
No i want for generator to work for seven seconds

**Comment by GreatSage777** [comment]
Its will be a reasonal buff for differential generator

**Comment by jehosula** [comment]
> No i want for generator to work for seven seconds

making it more complex does not solve the actual issue and instead will probably make it compete harder with later tech (which is bad btw)
if you want it to work for 7 seconds then make it take 2 pyratite (don't make it 7 seconds actually)
making it take spore just encourages skipping it as steam generator is more than enough to transition

**Comment by GreatSage777** [comment]
What about adding a new feature to differential generator like if you will input pyrotite, explosive substance and spores pod it will produce double energy.
2 pyrotite, 1 explosive substance , 2 spores pod and cyroflyid liquid. And differential generator will produce 2160 for 5 seconds

**Comment by jehosula** [comment]
saw the changes, and i think just make it so that the ultra mode increased production time by 2 seconds and does not increase pyratite and cryofluid inputs to make it worth
seems interesting now

**Comment by GreatSage777** [comment]
Sorry but if I am going to increase production time its will be too unbalanced so
For balance its better to decrease production time in ultra mode to: 5.00 -> 2.5
I dont agree with you about recources listen i just doubled the amount its generates 2160 for 2.5 seconds but it should use 2 pyrotite 1 explosive susbtance and 6/per second cryofluid input. Like just 1 pyrotite 1 explosive susbtance and just 2 cryofluid looking pretty op so therefore i doubled pyrotite input and make 3x cryofluid input.

**Comment by jehosula** [comment]
> Sorry but if I am going to increase production time its will be too unbalanced so For balance its better to decrease production time in ultra mode to: 5.00 -> 3.0 I dont agree with you about recources listen i just doubled the amount its generates 2160 for 3.0seconds but it should use 2 pyrotite 1 explosive susbtance and 6/per second cryofluid input. Like just 1 pyrotite 1 explosive susbtance and just 2 cryofluid looking pretty op so therefore i doubled pyrotite input and make 3x cryofluid input.

ultra mode will become worthless if it's not more efficient than normal mode

**Comment by GreatSage777** [comment]
Its more efficent now than safe mode

**Comment by GreatSage777** [comment]
Safe mode:
Production time: 5.0
Pyrotite: 1
Cryofluid input: 2/per second
Energy generating: 1080
Ultra mode: 
Production time: 3.00
Pyrotite: 2
Explosive substance: 1
Cryofluid input: 6.0
Energy generating: 2160

**Comment by GreatSage777** [comment]
Its just have to use some more pyrotite spores and cryofluid and you get doubled energy whats bad with it

**Comment by jehosula** [comment]
> Its more efficient now than safe mode

no, in fact, MUCH less efficient cause all the inputs, it's better to get 2 safe mode ones than one ultra mode one
0.2 pyratite --> 0.66 pyratite per second more than triple the pyratite
2 cryo --> 6 cryo triple the cryo
and also some blast compound, it's WAY too expensive for a booster style mechanic

**Comment by GreatSage777** [comment]
But if I am going to decrease the recources its will be too op for itso what should we do with ultra mode?

**Comment by jehosula** [comment]
> But if I am going to decrease the recources its will be too op for itso what should we do with ultra mode?

it should be a different kind of input that increased the output for cheap, it should be complexity vs efficiency, not cost + complexity vs output

**Comment by GreatSage777** [comment]
Maybe i should change pyrotite input in ultra mode to something else?

**Comment by jehosula** [comment]
> Maybe i should change pyrotite input in ultra mode to something else?

nah, keep the pyratite input when doing ultra mode, the activation for it should just be having blast compound in the generator which functions as a booster, and have that double the power output without increasing other inputs directly, this would work in making ultra mode produce more power for the cost while sacrificing simplicity

**Comment by GreatSage777** [comment]
But I done that
2 pyrotite 1 blast compound and 6/per second so what is your point

**Comment by jehosula** [comment]
> But I done that 2 pyrotite 1 blast compound and 6/per second so what is your point

my point is that it's too expensive to justify doing

**Comment by GreatSage777** [comment]
Okay 1 pyrotite 1 blast compund and maybe 4/cryo per second

**Comment by jehosula** [comment]
> Okay 1 pyrotite 1 blast compund and maybe 4/cryo per second

2 cryo, 
look at the water cooling for drills, cooling laser drills and airblasts with wex provides an actually good boost for production and for cheap, but takes a 2x2 square to set up (tbh that's not enough complexity)

**Comment by GreatSage777** [comment]
You mean 2/s?

**Comment by jehosula** [comment]
remember that blast compound is one spore + one pyratite, which is already more than 2 pyratite worth of inputs
even 1 pyratite and one blast compound and 2 cryo/s for ultra mode will struggle to compete with normal mode

**Comment by jehosula** [comment]
> You mean 2/s?

yes

**Comment by GreatSage777** [comment]
Ok i changed everything can you see more issues

**Comment by jehosula** [comment]
> Ok i changed everything can you see more issues

just remembered ultra mode can explode, so yeah just make it 2.5x power instead of 2x

**Comment by GreatSage777** [comment]
Emm 2160 for 5 seconds is already okay?

**Comment by jehosula** [comment]
> Emm 2160 for 5 seconds is already okay?

it has to be actively better in terms of production and efficiency, not equally good

**Comment by GreatSage777** [comment]
Ohhh its looking pretty op are you sure its better to change  this stat

**Comment by jehosula** [comment]
> Ohhh its looking pretty op are you sure its better to change this stat

in that case, just nerf the normal mode

**Comment by GreatSage777** [comment]
No normal mode is preety ok so you are right its better to upgrade ultra mode😊😊

**Comment by jehosula** [comment]
> No normal mode is preety ok so you are right its better to upgrade ultra mode😊😊

actually ~2.7k for something pre thorium is quite silly and it can be hit with an overdrive projector leading to a generator type block making 4k power
maybe it's the massive increase that is adding blast compound to the inputs?
in that case make it just 1.5x but make the blast compound input double the production time, so the inputs don't increase as much or maybe 9 seconds instead to make ultra mode less powerful but still good

**Comment by GreatSage777** [comment]
But you said more than 7 seconds is bad with later tech so you changed your mind?

**Comment by itcannotbe** [comment]
This entails making another generator "class" (dunno about the name if that's the proper one lol) as no generator supports "boost" items currently.

**Comment by GreatSage777** [comment]
Also i nerfed exploding 10 times because its too powerfull

**Comment by jehosula** [comment]
> But you said more than 7 seconds is bad with later tech so you changed your mind?

oh i forgot, anyway the issue i had was the complexity encouraging players to skip it like a lot do

**Comment by GreatSage777** [comment]
Em so ultra mode gives 1.5x enegy and what about production time?

**Comment by itcannotbe** [comment]
> > But you said more than 7 seconds is bad with later tech so you changed your mind?
> 
> oh i forgot, anyway the issue i had was the complexity encouraging players to skip it like a lot do

To be honest there is basically nothing we can do to encourage the usage of diff while not indirectly nerfing Impact.

**Comment by GreatSage777** [comment]
Maybe we are nerfing impact but do you want for forever using thorium or impact while not even knoeing that in the game are differential generator

**Comment by GreatSage777** [comment]
I want to make people start to look at differential generator not just to skip it

**Comment by GreatSage777** [comment]
Impact is good but its takes a time to turn on/off

**Comment by GreatSage777** [comment]
If something will not deliver to impact energy can die very easily

**Comment by itcannotbe** [comment]
Well, we can make the impact have a stupid long build time instead in order to force people to use Differential Generators.

**Comment by itcannotbe** [comment]
> Impact is good but its takes a time to turn on/off

It used to be worse, but that's a long time ago.

**Comment by GreatSage777** [comment]
We are here to talk about diff not impact actually because in thus issues i buffing diff not impact

**Comment by jehosula** [comment]
pretty sure diffgen competes with thorium reactor more than impact reactor cause of the tech level

**Comment by itcannotbe** [comment]
> pretty sure diffgen competes with thorium reactor more than impact reactor cause of the tech level

It does, and it doesn't explode too.

**Comment by jehosula** [comment]
maybe instead of all that an idea could be to make it either not use pyratite or make it take water instead of cryofluid

**Comment by jehosula** [comment]
or just nerf steam gen so it becomes harder to skip diffgen

**Comment by GreatSage777** [comment]
**you dont understand? Tha main point in diff is that he uses the difference between cryo and pyrotite you just want to destroy the main point**

**Comment by itcannotbe** [comment]
> maybe instead of all that an idea could be to make it either not use pyratite or make it take water instead of cryofluid

That's too extreme.

**Comment by GreatSage777** [comment]
I think i already baffed it enough so safe mode is pretty safe and good but ultra mode is much better

**Comment by jehosula** [comment]
> **you dont understand? Tha main point in diff is that he uses the difference between cryo and pyrotite you just want to destroy the main point**

yes the idea was swapping the inputs for something simpler, to make it easier to get it was not that extreme
also what does blast compound do?

**Comment by GreatSage777** [comment]
Its boosting diff?

**Comment by jehosula** [comment]
> Its boosting diff?

but how? it's just explosive, not flammable or cold

**Comment by GreatSage777** [comment]
And changes between ultra and safe mode

**Comment by GreatSage777** [comment]
Yea like its boosting diff itself to make it work faster

**Comment by itcannotbe** [comment]
> or just nerf steam gen so it becomes harder to skip diffgen

This is the worse idea, we don't change steam gen stats for something mundane.

**Comment by EvDes190** [comment]
how about buffing output energy and input cryo at hot surfaces
